sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -791,53 +805,69 @@ private BasicDocumentRevision deleteDocumentInQueue(SQLDatabase db, final String
-                        if (preRevision == null) {
-                            throw new IllegalArgumentException("The document trying to update does not exist.");
-                        }
 
-                        DocumentRevisionTree revisionTree = getAllRevisionsOfDocumentInQueue(db,
-                                docId);
-                        if(revisionTree == null) {
-                            throw new IllegalArgumentException("Document does not exist for id: " + docId);
-                        } else if (!revisionTree.leafRevisionIds().contains(prevRevId)) {
-                            throw new ConflictException("Revision to be deleted is not a leaf node:"
-                                    + prevRevId);
-                        }
+        DocumentRevisionTree revisionTree;
+        try {
+            revisionTree = getAllRevisionsOfDocumentInQueue(db, docId);
+        } catch(AttachmentException e){
+            // we can't get load the document, due to an attachment error,
+            // throw an exception saying we couldn't find the document.
+            throw new DocumentNotFoundException(e);
+        }
 
-                        if(!preRevision.isDeleted()) {
-                            setCurrent(db,preRevision, false);
-                            String newRevisionId = CouchUtils.generateNextRevisionId(preRevision.getRevision());
-                            // Previous revision to be deleted could be winner revision ("current" == true),
-                            // or a non-winner leaf revision ("current" == false), the new inserted
-                            // revision must have the same flag as it previous revision.
-                            // Deletion of non-winner leaf revision is mainly used when resolving
-                            // conflicts.
-                            InsertRevisionOptions options = new InsertRevisionOptions();
-                            options.docNumericId = preRevision.getInternalNumericId();
-                            options.revId = newRevisionId;
-                            options.parentSequence = preRevision.getSequence();
-                            options.deleted = true;
-                            options.current = preRevision.isCurrent();
-                            options.data = JSONUtils.EMPTY_JSON;
-                            options.available = false;
-                            options.copyAttachments = false;
-                            insertRevision(db,options);
-                            deletedRevision = getDocumentInQueue(db, preRevision.getId(),
-                                    newRevisionId);
-                            documentDeleted = new DocumentDeleted(preRevision, deletedRevision);
-                        }
+         if (!revisionTree.leafRevisionIds().contains(prevRevId)) {
+            throw new ConflictException("Revision to be deleted is not a leaf node:"
+                    + prevRevId);
```

Comment:
Say something that's more meaningful to a developer, like "Document has newer revisions than the revision passed to delete".